test/lib/object-diff.js

```diff
@@ -0,0 +1,24 @@
+var assert = require('assert')
+var difference = require('../../src/lib/object-diff')
+
+describe('object difference', function() {
+  it('should return difference', function() {
+    assert.deepEqual(
+      difference({a: ['a', 'b', 'c']}, {a: ['x', 'y', 'z']}),
+      [['a', ['a', 'b', 'c'], ['x', 'y', 'z']]]
```

Comment:
I guess the `diff` schema would be better if its explicit
```
[
  ...
  { 
    rule : {
      config1: config1, 
      config2: config2
    } 
  }
  ...
]
```

instead of position based

```
[['a', ['a', 'b', 'c'], ['x', 'y', 'z']]]
```